homeassistant/components/telnet/switch.py

```diff
@@ -149,6 +149,7 @@ def update(self):
         if response:
             rendered = self._value_template.render_with_possible_json_value(response)
             self._state = rendered == "True"
+            self.async_schedule_update_ha_state()
```

Comment:
This is a sync method, it should not call async methods in here.
Use `schedule_update_ha_state()` instead.

Reply:
If the entity is polling this shouldn't be needed. There will already be a state update scheduled for polling entities.

Reply:
True, @zentale, how big is the delay in this case that this is a concern?

Reply:
I click the button on the UI, it immediately turns on my lamp. Still I wait 5-10 sec for the UI to refresh via poll. Making the changes mentioned.

Reply:
Should be fine now, removed the copy from the update method, and using the sync version now of the method.

Reply:
The point being, it should poll immediately after changing, doesn't it do that?

Reply:
It doesn't poll immediately. I think his concern is only for the update inside the update - which is probably useless indeed.

Reply:
It won't poll for the assumed state case (poll turned off). So explicitly scheduling in the action is correct.